### ISSUE TITLE
Rescue HUD isn't updated until mouseexit

### ISSUE THREAD

**Issue by esheyw** [xaukael/covered-token-rescue-hud#16]
![](https://i.imgur.com/yHB2CPB.gif)

As long as the mouse is over a rescue hud token image, the list of hidden tokens isn't updated

**Comment by uniwars** [comment]
Same.

**Comment by xaukael** [comment]
That was an issue that came up due to hover selectors. I might be able to come up with a better solution, perhaps just refreshing every time it is clicked, though that will surely introduce some other weird behavior.

**Comment by xaukael** [comment]
I changed it to make it update when a token is clicked. That should resolve this issue?

**Comment by uniwars** [comment]
yes it seems to work fine, idk if it's new or not but as GM when clicking on a mini token it will select the actor, and when right clicking it will also select it, so it's impossible to target a hidden token without also selecting it

**Comment by esheyw** [comment]
![](https://i.imgur.com/2ImiT8d.gif)

This (current release of the module behaviour) is objectively better, but still a couple notes:
1. first in the gif, you can see that now the hud doesn't go away when the mouse isn't hovering anything in the stack, until a different token/stack is hovered. not a huge problem, but would be cleaner if it hid on mouseexit from the stack, if that's determinable
2. Like uniwars noted, there's some weirdness with right clicking. sometimes it opens the token hud for the top of the stack, sometimes for the rescue hud token. Personally, I think you should eat right clicks and reject them entirely. doing something to the token should consist of bringing it to the top first, then do stuff. This is a philosophical position others might not agree with though 🤷‍♂️
